### ISSUE TITLE
Amplify applied multiple times in 1.4.0-alpha1

### ISSUE THREAD

**Issue by toots** [savonet/liquidsoap#717]
From the mailing list:

In Liquidsoap 1.3.6, the
"liq_amplify" annotation works as expected: the track level is
increased by exactly the amount given, whether or not the playlist is
passed through the crossfade function.

But in Liquidsoap 1.4.0-alpha1, the "liq_amplify" annotation, if
present, appears to result in much more gain than what is specified.
If it is absent, or set to 0, there is no change in gain.

Here is my test playlist of three short tracks. Each is recorded
quietly, so each requires an amount of gain. These are three extreme
examples, but all are measured properly.

```
annotate:liq_cue_in="0.000",liq_start_next="5.560",duration="312.060",liq_amplify="8.000dB":/home/john/src/radio/mez/204-zoot_sims-blue_room.PbTVdK.mka
annotate:liq_cue_in="0.800",liq_start_next="9.140",duration="286.840",liq_amplify="8.200dB":/home/john/src/radio/mez/02
Cha Cha Loco.FKMSrM.mka
annotate:liq_cue_in="0.100",liq_start_next="5.280",duration="162.080",liq_amplify="8.100dB":/home/john/src/radio/mez/12.Song
to Woody.wOgkma.mka
```

These two lines under the release version produce correct results:

```
myplaylist = amplify(override="liq_amplify", 1.0,
cue_cut(playlist(length=60.0, mode="normal",
"/home/john/src/radio/cha.m3u8")))
myplaylist = crossfade(fade_out=0.01, fade_in=0.01, conservative=true,
 myplaylist)
```

But under 1.4.0-alpha1, the addition of the second line produces way
too much gain. Under 1.3.6, it produces the correct amount of gain.

When I did the tests, the "Overriding amplification" notes appeared. The numbers seemed right, but they appeared three times. Under version 1.3.6, they appear only once.

e.g., in version 1.4.0:

```
2019/02/03 08:49:59 [cha(dot)m3u8:3] Prepared "/home/john/src/radio/mez/204-zoot_sims-blue_room.PbTVdK.mka" (RID 2).
2019/02/03 08:49:59 [fallback_10413:3] Switch to cross_10408 with transition.
2019/02/03 08:49:59 [single_10392:4] Activations changed: static=[], dynamic=[fallback_10413:Music_Too:Music_Too].
2019/02/03 08:49:59 [cross_10408:4] Activations changed: static=[fallback_10413:Music_Too:Music_Too], dynamic=[fallback_10413:Music_Too:Music_T
oo].
2019/02/03 08:49:59 [cross_10408:4] Buffering end of track...
2019/02/03 08:49:59 [amplify_10399:3] Overriding amplification: 2.511886.
2019/02/03 08:49:59 [amplify_10399:4] Fading in..
2019/02/03 08:49:59 [fade(dot)in:3] Overriding amplification: 2.511886.
2019/02/03 08:49:59 [fade(dot)in:4] Fading out with 0.0s remaining..
2019/02/03 08:49:59 [fade(dot)out:3] Overriding amplification: 2.511886.
2019/02/03 08:49:59 [decoder:4] Trying method "META" for "/home/john/src/radio/mez/02 Cha Cha Loco.FKMSrM.mka"...
2019/02/03 08:49:59 [decoder:4] Trying method "WAV" for "/home/john/src/radio/mez/02 Cha Cha Loco.FKMSrM.mka"...
```

The log above was made during severe distortion on the tracks amplified. Taking out the amplification, or the cross-fade, resulted in no distortion.

Normally, the "Overriding amplification" notices appear only once. Here's the last few minutes of my present log using version 1.3.6:

```
2019/02/07 17:29:13 [amplify_10226:3] Overriding amplification: 0.749894.
2019/02/07 17:29:13 [decoder:3] Method "FFMPEG" accepted "/home/john/src/radio/mez/08. barrion nuevo.nHoocd.mka".
2019/02/07 17:33:26 [test-nodup-shuf-processed(dot)m3u8:3] Finished with "/home/john/src/radio/mez/Orinoco_Flow_Enya.SjoVpS.mka".
2019/02/07 17:33:26 [amplify_10226:3] End of the current overriding.
2019/02/07 17:33:26 [test-nodup-shuf-processed(dot)m3u8:3] Prepared "/home/john/src/radio/mez/08. barrion nuevo.nHoocd.mka" (RID 3).
2019/02/07 17:33:26 [amplify_10226:3] Overriding amplification: 1.035142.
2019/02/07 17:33:27 [decoder:3] Method "FFMPEG" accepted "/home/john/src/radio/mez/01 Tuane.MMCDIy.mka".
2019/02/07 17:36:01 [server:3] New client: 127.0.0.1.
2019/02/07 17:36:01 [server:3] Client 127.0.0.1 disconnected.
2019/02/07 17:36:01 [decoder:3] Method "FFMPEG" accepted "/home/john/src/radio/bbcuknews.mka".
2019/02/07 17:39:17 [test-nodup-shuf-processed(dot)m3u8:3] Finished with "/home/john/src/radio/mez/08. barrion nuevo.nHoocd.mka".
2019/02/07 17:39:17 [amplify_10226:3] End of the current overriding.
2019/02/07 17:39:17 [override:3] Prepared "/home/john/src/radio/bbcuknews.mka" (RID 5).
2019/02/07 17:39:17 [fallback_10229:3] Switch to override with forgetful transition.
2019/02/07 17:42:19 [override:3] Finished with "/home/john/src/radio/bbcuknews.mka".
2019/02/07 17:42:19 [test-nodup-shuf-processed(dot)m3u8:3] Prepared "/home/john/src/radio/mez/01 Tuane.MMCDIy.mka" (RID 4).
2019/02/07 17:42:19 [fallback_10229:3] Switch to amplify_10226 with forgetful transition.
2019/02/07 17:42:19 [amplify_10226:3] Overriding amplification: 0.749894.
```

**Comment by dhannyz** [comment]
@toots i do a somewhat more advanced LUFS based determination of what my gains should be. Setting my own custom meta tags and reading them back in, and using that for the gain.
(perhaps this could be a new feature, as its a relatively easy ffmpeg filter)
However I am not getting the the above issue, perhaps that narrows it down for you?

**Comment by toots** [comment]
Hey! Thanks for chiming in. The `fade` operators are deeply rewritten in this alpha branch and I'm suspecting that's where the issue comes from.

**Comment by savonet-users** [comment]
On Thu, Feb 21, 2019 at 4:25 PM Romain Beauxis <notifications@github.com>
wrote:

> Hey! Thanks for chiming in. The fade operators are deeply rewritten in
> this alpha branch and I'm suspecting that's where the issue comes from
>
Indeed, the same is still happening (very loud amplification) on the latest
git master (is there another branch I should be testing?) No worries. Will
wait.

(P.S., the overlapping again died this morning after about six days of
continuous playout, at the same time as any lines mentioning "buffer" or
"buffering" vanished from the debug log. This is what it did last week,
too. The log didn't mention any errors. Are there any extra debug flags I
can set to produce output which may be more useful to you?)

with best wishes,
JW

**Comment by toots** [comment]
Thanks for reporting this one guys. That was easy to fix an typically the kind of things that's great to find out in early testing phases!

**Comment by savonet-users** [comment]
On Thu, Feb 28, 2019 at 1:44 AM Romain Beauxis <notifications@github.com>
wrote:

> Thanks for reporting this one guys. That was easy to fix an typically the
> kind of things that's great to find out in early testing phases!
>
Thanks for looking into this—and fixing it!

I'm now on git master and it's been up for about 24 hours so far; and the
problem with crossfades permanently disappearing after an unexpected time
hasn't happened yet. This used to take between one and six days to manifest
itself, so I'll keep running and, if it happens, will raise another issue
and post logs, scripts, etc.

with best wishes,
JW
